src/iceberg/avro/avro_schema_util_internal.h

```diff
@@ -163,4 +163,9 @@ Result<::avro::NodePtr> MakeAvroNodeWithFieldIds(const ::avro::NodePtr& original
 Result<::avro::NodePtr> MakeAvroNodeWithFieldIds(const ::avro::NodePtr& original_node,
                                                  const NameMapping& mapping);
 
+/// \brief Sanitize a field name to make it compatible with Avro field name requirements.
```

Comment:
Could you please also add a `bool ValidateFieldName(std::string_view)`?

Reply:
In addition, it would be good to explicitly state what are the `Avro field name requirements` in the comment.